tensorflow/c/kernels.cc

```diff
@@ -551,3 +571,301 @@ TF_Tensor* TF_AllocateTemp(TF_OpKernelContext* context, TF_DataType dtype,
   }
   return tf_tensor;
 }
+
+tensorflow::Status EnsureSparseVariableAccess(TF_OpKernelContext* ctx,
```

Comment:
Is it possible to share code with the existing impl for `EnsureSparseVariableAccess` in `training_op_helpers.h`? Same comment for other helpers below.

Reply:
Thanks @saxenasaurabh for the review. I have used the helper functions where possible like `LookupResource`. With [EnsureSparseVariableAccess](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/kernels/training_op_helpers.h;l=31) , there is [Device dependency](https://cs.opensource.google/tensorflow/tensorflow/+/master:tensorflow/core/kernels/training_op_helpers.h;l=62) which we are passing in as Copy functors. I do agree there is duplication of code which can be avoided.  One possibility is to refactor the core helper functions to remove this dependency and we can adopt it here.  Maybe we can do it as a followup cleanup as it will require invasive changes in the core which can break things. What do you think ?

Reply:
Sounds good. Please try to clean this is up as a follow-up. That would give test coverage for free as well.

Reply:
Makes sense, will do.